core/server/master/src/main/java/alluxio/master/service/metrics/WhenLeadingMetricsSimpleService.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.master.service.metrics;
+
+class WhenLeadingMetricsSimpleService extends MetricsSimpleService {
```

Comment:
For these new services, could you add some basic comments here please. Like to say that they are created by {@link SomeFactory} in the case of {X}. And that it is different when leading vs not leading because it does {X} differently.